Hints.py

```diff
@@ -114,73 +114,80 @@ def isRestrictedDungeonItem(dungeon, item):
     return False
 
 
-def add_hint(spoiler, world, IDs, gossip_text, count, location=None, force_reachable=False):
-    random.shuffle(IDs)
+def add_hint(spoiler, world, groups, gossip_text, count, location=None, force_reachable=False):
+    random.shuffle(groups)
     skipped_ids = []
     duplicates = []
     first = True
     success = True
     # early failure if not enough
-    if len(IDs) < int(count):
+    if len(groups) < int(count):
         return False
-    # Randomly round up, if we have enough IDs left
-    total = int(random.random() + count) if len(IDs) > count else int(count)
+    # Randomly round up, if we have enough groups left
+    total = int(random.random() + count) if len(groups) > count else int(count)
     while total:
-        if IDs:
-            id = IDs.pop(0)
+        if groups:
+            group = groups.pop(0)
 
-            if gossipLocations[id].reachable:
-                stone_name = gossipLocations[id].location
-                stone_location = world.get_location(stone_name)
-                if not first or can_reach_stone(spoiler.worlds, stone_location, location):
+            if any(map(lambda id: gossipLocations[id].reachable, group)):
+                stone_names = [gossipLocations[id].location for id in group]
+                stone_locations = [world.get_location(stone_name) for stone_name in stone_names]
+                if not first or any(map(lambda stone_location: can_reach_stone(spoiler.worlds, stone_location, location), stone_locations)):
                     if first and location:
                         # just name the event item after the gossip stone directly
-                        MakeEventItem(stone_name, stone_location)
+                        for i, stone_name in enumerate(stone_names):
+                            MakeEventItem(stone_name, stone_locations[i])
+
                         # This mostly guarantees that we don't lock the player out of an item hint
                         # by establishing a (hint -> item) -> hint -> item -> (first hint) loop
-                        location.add_rule(world.parser.parse_rule(repr(stone_name)))
+                        group_rule = ' or '.join(stone_names)
```

Comment:
Why not just make a single event item to place at all the stones in the group?

Reply:
That makes a lot more sense. I wasn't sure how exactly the event item constructor worked so I wasn't sure how hard it would be to have multiple locations with the same event item.